### ISSUE TITLE
Star Citizen 3.3 ships transform matrix

### ISSUE THREAD

**Issue by Freeman711** [Markemp/Cryengine-Converter#18]
Hello!
Thanks for your project.

Unfortunately, I encountered a number of problems when importing sc 3.3 files.
1. Converter version 1.03 creates files from which you can import only the skeleton. The geometry is present in the file, but not processed. It looks like an error in the structure of xml. Tested on a blender, autodesk collada and opencollada
2. Most models of ships have problems with the transform matrix - their parts are not in their places, turned, etc.

Sample files are ready to send to you say.
Also posted them in your topic on xentax 
http://forum.xentax.com/viewtopic.php?f=33&t=13137&sid=cb7c3dded970564401baa9cf2aedb66e&start=90

Can you help?
Thank you.

**Comment by Loganbacca** [comment]
I've also encountered this.

Geometry loads ok from .obj files, but not from .dae files.
Separate parts in the model are also not aligned correctly.

**Comment by Tercerobootleg** [comment]
I'm having the issue with parts being unaligned with .obj as well.  Would love to see a fix for this.

**Comment by vmxeo** [comment]
Just a bump on this, as I'm running into this issue as well. Seems during conversion to for any geometry that's parented. OBJs don't get the part position, and DAE doesn't include the child geometry (I hope that makes sense). I can provide files as examples if it helps.

**Comment by Markemp** [comment]
Exporting to OBJ isn't supported anymore.  Just too many required features that it can't support (such as object parenting), unfortunately.  

Star Citizen is in a weird place where a bunch of us in the modding community are on a bit of hiatus until they lock down the structures.  The unpacker tools change almost every release, along with the model files.  Once they settle down on the changes, we'll get back into making sure SC is fully supported though.

I'll leave this issue open until that happens as a reminder to follow up with everyone on this thread.

**Comment by draakuns** [comment]
Hi @Markemp,
thanks for you hard work on this topics!

Collada files (on SC) are not working on 1.0.4 yet, but OBJ are (with their lack of order and stuff, but hey...). Is it possible to get the info from the OBJs to Collada in an (relatively) easy way?
I dont mean just you coding, but at least something that can help us to get some models to tinker in the meanwhile or as a workaround.
Or if we can provide some help on it, (comparing xmls/formats or other stuff) just let me know.

Best regards,
Draakuns

**Comment by vmxeo** [comment]
@draakuns As a workaround, I've been converting to *both* dae and obj, and made a script to import and join both back together. It's not perfect, but works well enough.

**Comment by draakuns** [comment]
@vmxeo thanks, I will try to find middle grounds for this, until time to fix this is available for any of us.

**Comment by richardthombs** [comment]
Hi there,

I’m trying to find a way of creating silhouettes of star citizen ships – kind of like the old fighter & ship identification charts. Although I’m a programmer, I know nothing about 3d modelling. I’ve discovered this repo and I’ve converted some of the SC .cga files to Collada files, but when I import them into blender, I don’t see a ship model. One file I converted seemed to give the 3d position of the various ship parts, another seemed to give a set of parts at a certain level of detail, but all the parts were on top of each other.

Does anyone have any idea how to turn these into a proper model please? Or can you point me in the right direction?

Thank you!

R.

**Comment by vmxeo** [comment]
@richardthombs My workaround is to convert _both_ DAE(for position) and obj(for geometry). I made a script that then merges them back together. It works, but it's not perfect.

**Comment by vmxeo** [comment]
@Markemp 

Hey, I think I figured out the problem we're having with SC geometry not converting in DAE files. The XML is correct, and the data is there, but...
```
<library_visual_scenes>
    <visual_scene id="Scene">
      <node id="ARGO_MPUV" name="ARGO_MPUV" type="NODE">
        <matrix sid="transform">1.000000 0.000000 0.000000 0.000000 0.000000 1.000000 0.000000 0.000000 0.000000 0.000000 1.000000 0.000000 0.000000 0.000000 0.000000 1.000000</matrix>
        <node id="Nose" name="Nose" type="NODE">
          <matrix sid="transform">1.000000 0.000000 0.000000 0.000000 0.000000 1.000000 0.000000 3.970069 0.000000 0.000000 1.000000 -0.674325 0.000000 0.000000 0.000000 1.000000</matrix>
```
If I'm reading this right, the visual scene node is setting up a matrix for some geometry, but never actually instancing an existing geometry node to it. I added the following to the dae file right after the matrix node:
`		  <instance_geometry url="#Nose-mesh" name="Nose"/>`

and it works. 

Hope that's clear enough explanation. I'd go through the code and submit a change, but I'm just a python monkey, and wouldn't know where to start :)

**Comment by wpw** [comment]
@vmxeo 

I just tried this last night with the Terrapin and it definitely works. Just adding another "can reproduce" to the pile. All transforms are correct (unlike .obj) and the meshes are good. I think some of the parenting/hierarchy might be wonky, some parts are children of other objects which don't always seem logical, but that may be the case with the original source file and not a problem with the converter.

Considering writing my own "fixer" script in the interim to insert the appropriate instance_geometry tags because doing it by hand on a medium sized ship is less than ideal, to put it lightly.

**Comment by Markemp** [comment]
Yup.  This is closely related to issue #32 as well.  I'm migrating my website this weekend, but will pick this back up as soon as it's done.  It shouldn't take too long to have a fixed version released.

**Comment by coloradocomputerpro** [comment]
I can vouch this has been resolved with 1.0.5

**Comment by Markemp** [comment]
Perfect!  Thank you for your help again.  Closing this issue.
